testing/README.md

```diff
@@ -0,0 +1,95 @@
+  - `kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/v1.5.4/cert-manager.yaml`
+  - `curl -Lo v2_4_7_full.yaml https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/download/v2.4.7/v2_4_7_full.yaml`
+  - `sed -i.bak -e '561,569d' ./v2_4_7_full.yaml`
+  - `sed -i.bak -e 's|your-cluster-name|<ClusterName>|' ./v2_4_7_full.yaml`
+  - `kubectl apply -f v2_4_7_full.yaml`
+  - `curl -Lo v2_4_7_ingclass.yaml https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/download/v2.4.7/v2_4_7_ingclass.yaml`
+  - `kubectl apply -f v2_4_7_ingclass.yaml`
+
+### 3. Setting up Environment for EC2 Tests
+#### Create IAM Role for EC2 Instance
+- Login to AWS, go to the IAM console and click on the Roles tab. Click Create role.
+- Choose AWS service, and choose EC2 as the use case. Click Next.
+- Choose AmazonS3ReadOnlyAccess, AWSXrayWriteOnlyAccess, and CloudWatchAgentServerPolicy as the permission. 
+- Type the role name and click "Create role".
+- 
+#### Setting Up Default VPC
+- Go to the VPC console and on the routing table for the default VPC, click Edit routes. 
+- Click add routes, for destination add `0.0.0.0/0`, for target add Internet Gateway and save changes.
+- Go to the Security groups tab, find the security group attached to the default VPC, click Edit inbound rules, choose type: All Traffic, Source: custom, and CIDR block: 0.0.0.0/0. Save rules.
+
+### 4. Building Sample App to ECR
+Create two ECR repositories: one for the sample app main service and another for the sample app remote service. 
+Follow the instructions under ./sample-apps/README.md to build the sample app image and upload it to the ECR
+
+### 5. Building Sample App to S3 Bucket
+Create an S3 Bucket to store the .jar files for the sample app main service and sample app remote service.
+Follow the instructions under ./sample-apps/README.md to build the sample app .jar and upload it to the bucket
+
+### 6. Setting up repository
+- Go to https://github.com/aws-observability/aws-otel-java-instrumentation and create a fork
+- Go to the forked repo and enable action on the Action tab
+- Add the following secrets to the repository
+  - APP_SIGNALS_E2E_TEST_ACC: `<AccountID>`
+  - E2E_TEST_ROLE_ARN: `arn:aws:iam::<AccountID>:role/<RoleName>`
+  - APP_SIGNALS_E2E_EC2_TEST_ROLE: <EC2_IAM_ROLE_NAME>
+  - APP_SIGNALS_E2E_FE_SA_IMG: `<AccountID>.dkr.ecr.us-east-1.amazonaws.com/<Path to Sample App Image>`
+  - APP_SIGNALS_E2E_RE_SA_IMG: `<AccountID>.dkr.ecr.us-east-1.amazonaws.com/<Path to Remote Sample App Image>`
+  - APP_SIGNALS_E2E_FE_SA_JAR: s3://<BucketName>/<FileName.jar>
+  - APP_SIGNALS_E2E_RE_SA_JAR: s3://<BucketName>/<FileName.jar>
+
+
+### 7. Running the tests
+Copy paste the test.yml into `../.github/workflows` and replace the cluster name with the one generated in step 2. 
+Push the code changes and there should be a test running on the forked repo in the Action tab
```

Comment:
The tester should just be able to run the canary on their test environment. Is that doable without having to make edits?

Reply:
Is that what we want? Rather than having a canary running on their forked repo isn't it better to just test change when we push new commit? Or do you mean we change the `on:` value to push on the canary file when we want to test?

Reply:
It would be best if the tester doesn't have to add files and then remove them after the fact. If we're setting up a test environment it should be made to be exactly like the real environment so that we can just run the same test in our accounts.